From df91d0484e97c099291ce787f93b1d1ce879f26a Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Tue, 23 Apr 2024 15:32:58 +0530
Subject: [PATCH] mmc: sdhci: Add kzalloc to allocate ice_config_sec struct
 instead of kmalloc

Add fix to allocate ice_config_sec structure using kzalloc instead of kmalloc.
Kzalloc use to initialize structure vailable as zero, it
use to avoid garbage values in structure members.

Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Change-Id: I9b67c27be93d67bf609eb3c27c5ea9ff8fcb8743
---
 drivers/mmc/host/sdhci-msm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-msm.c b/drivers/mmc/host/sdhci-msm.c
index 8fdb881160ab..f3ce9b451ad6 100644
--- a/drivers/mmc/host/sdhci-msm.c
+++ b/drivers/mmc/host/sdhci-msm.c
@@ -2063,7 +2063,7 @@ static int sdhci_msm_ice_set_hwkey_config(struct cqhci_host *cq_host,
 	struct ice_config_sec *ice = NULL;
 	int ret;
 
-	ice = kmalloc(sizeof(struct ice_config_sec), GFP_KERNEL);
+	ice = kzalloc(sizeof(struct ice_config_sec), GFP_KERNEL);
 	if (!ice)
 		return -ENOMEM;
 
-- 
2.34.1

